modules/common/tls/tls.go

```diff
@@ -45,133 +54,417 @@ const (
+			hash, ctrlResult, err := endpointTLSCfg.ValidateCertSecret(ctx, h, namespace)
+			if err != nil {
+				return "", ctrlResult, err
+			} else if (ctrlResult != ctrl.Result{}) {
+				return "", ctrlResult, nil
+			}
+
+			certHashes["cert-"+endpt.String()] = env.SetValue(hash)
+		}
+	}
+
+	certsHash, err := util.HashOfInputHashes(certHashes)
+	if err != nil {
+		return "", ctrl.Result{}, err
+	}
+	return certsHash, ctrl.Result{}, nil
+}
+
+// getCertMountPath - return certificate mount path
+func (s *Service) getCertMountPath(serviceID string) string {
+	if serviceID == "" {
+		serviceID = "default"
 	}
 
-	if t.Ca != nil && t.Ca.CaSecretName != "" {
+	certMountPath := fmt.Sprintf("%s/%s.crt", DefaultCertMountDir, serviceID)
+	if s.CertMount != nil {
+		certMountPath = *s.CertMount
+	}
+
+	return certMountPath
+}
+
+// getKeyMountPath - return key mount path
+func (s *Service) getKeyMountPath(serviceID string) string {
+	if serviceID == "" {
+		serviceID = "default"
+	}
+
+	keyMountPath := fmt.Sprintf("%s/%s.key", DefaultKeyMountDir, serviceID)
+	if s.KeyMount != nil {
+		keyMountPath = *s.KeyMount
+	}
+
+	return keyMountPath
+}
+
+// CreateVolumeMounts - add volume mount for TLS certificates and CA certificate for the service
+func (s *Service) CreateVolumeMounts(serviceID string) []corev1.VolumeMount {
+	volumeMounts := []corev1.VolumeMount{}
+	if serviceID == "" {
+		serviceID = "default"
+	}
+	if s.SecretName != "" {
 		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      "ca-certs",
-			MountPath: "/etc/pki/ca-trust/extracted/pem",
+			Name:      serviceID + "-tls-certs",
+			MountPath: s.getCertMountPath(serviceID),
+			SubPath:   CertKey,
+			ReadOnly:  true,
+		}, corev1.VolumeMount{
+			Name:      serviceID + "-tls-certs",
+			MountPath: s.getKeyMountPath(serviceID),
+			SubPath:   PrivateKey,
 			ReadOnly:  true,
 		})
+
+		if s.CaMount != nil {
+			volumeMounts = append(volumeMounts, corev1.VolumeMount{
+				Name:      serviceID + "-tls-certs",
+				MountPath: *s.CaMount,
+				SubPath:   CAKey,
+				ReadOnly:  true,
+			})
+		}
 	}
 
 	return volumeMounts
 }
 
-// CreateVolumes - add volume for TLS certificate and CA certificates
-func (t *TLS) CreateVolumes() []corev1.Volume {
-	var volumes []corev1.Volume
-
-	if t.Service != nil && t.Service.SecretName != "" {
-		volumes = append(volumes, corev1.Volume{
-			Name: "tls-certs",
+// CreateVolume - add volume for TLS certificates and CA certificate for the service
+func (s *Service) CreateVolume(serviceID string) corev1.Volume {
+	volume := corev1.Volume{}
+	if serviceID == "" {
+		serviceID = "default"
+	}
+	if s.SecretName != "" {
+		volume = corev1.Volume{
+			Name: serviceID + "-tls-certs",
 			VolumeSource: corev1.VolumeSource{
 				Secret: &corev1.SecretVolumeSource{
-					SecretName:  t.Service.SecretName,
-					DefaultMode: ptr.To[int32](0440),
+					SecretName:  s.SecretName,
+					DefaultMode: ptr.To[int32](0400),
```

Comment:
If we want to use this without root (using FSGroup) does this need to be 0440?

Reply:
yes, maybe. for the services we did so far, non uses non root without kolla. if needed we can introduce a `*DefaultMode` to the Service type so that it could be customized if needed

Reply:
ack, will leave this for follow-up patch if/when it's required